packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -531,7 +531,7 @@ export class ContextualMenuBase extends BaseComponent<IContextualMenuProps, ICon
 
     if (section.items && section.items.length > 0) {
       return (
-        <li role="presentation" key={section.key}>
+        <li role="presentation" key={`section-${index}`}>
```

Comment:
Not completely sure if this is the correct fix. I would be more inclined to fix the example not to use the same key. If thinking I would even say that no one should actually use identic keys in their code when providing an array of items. Better maybe fix the example and in here do something like this to account for the case when someone actually provided items with the same key.
```suggestion
        <li role="presentation" key={`section-${section.key}-${index}`}>
```

Reply:
Neat idea @Vitalius1! However, my instinct is the same as @Vitalius1's original suggestion which is to _only_ fix the example code. This will benefit others who refer to it and face the error you did @danmarshall.

My understanding is that using `index` as a `key` should be avoided, which _I think_ would be the case here if `section.key` were non-unique or `undefined` (i.e. `section-undefined-0`) and the underlying array was modified.

Is my understanding correct or am I overthinking? Can we test for this case?

What's interesting is that React would default to using `index` here if we didn't provide the `section-` prefix.

- https://reactjs.org/docs/lists-and-keys.html#keys
- https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Thanks for the thoughtful discussion 🍻! So, there is an actual bug here which prevents us from using `section.key`. Looking at [line 517](https://github.com/OfficeDev/office-ui-fabric-react/blob/63455f22e5555d869c9b59e19434407ce947195e/packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx#L517) :
```js
const section = item.sectionProps
```
this variable should actually be named `sectionProps` since that is what it is in the json. Taking another look at the json:
```
{
  key: 'section',
  itemType: ContextualMenuItemType.Section,
  sectionProps: {
    //key does not exist here under sectionProps
    topDivider: true,
    bottomDivider: true,
    title: 'Actions',
    items: [
      {
        key: 'newItem',
        text: 'New'
      },
      {
        key: 'deleteItem',
        text: 'Delete'
      }
    ]
  }
}
```
We see the bug and why `key` is undefined.

Reply:
I've renamed the variables to better represent their types, can you re-review ? Hopefully you can see the fix which is to use `sectionItem.key`, and also to change the examples to have unique keys.